packages/core/src/types/arazzo.ts

```diff
@@ -0,0 +1,386 @@
+    stepId: { type: 'string' },
+    workflowId: { type: 'string' },
+    criteria: criteriaObjects,
+  },
+  additionalProperties: false,
+  required: ['type', 'name'],
+} as const;
+const onSuccessList = {
+  type: 'array',
+  items: onSuccessObject,
+} as const;
+const onFailureObject = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    type: { type: 'string', enum: ['goto', 'retry', 'end'] },
+    workflowId: { type: 'string' },
+    stepId: { type: 'string' },
+    retryAfter: { type: 'number' },
+    retryLimit: { type: 'number' },
+    criteria: criteriaObjects,
+  },
+  additionalProperties: false,
+  required: ['type', 'name'],
+} as const;
+const onFailureList = {
+  type: 'array',
+  items: onFailureObject,
+} as const;
+const step = {
+  type: 'object',
+  properties: {
+    stepId: { type: 'string' },
+    description: { type: 'string' },
+    operationId: { type: 'string' },
+    operationPath: { type: 'string' },
+    workflowId: { type: 'string' },
+    parameters: parameters,
+    successCriteria: criteriaObjects,
+    onSuccess: onSuccessList,
+    onFailure: onFailureList,
+    outputs: {
+      type: 'object',
+      additionalProperties: {
+        type: 'string',
+      },
+    },
+    'x-inherit': inherit,
+    'x-expect': expectSchema,
+    'x-assert': { type: 'string' },
+    'x-operation': extendedOperation,
+    requestBody: requestBody,
+  },
+  required: ['stepId'],
+  oneOf: [
+    { required: ['x-operation'] },
+    { required: ['operationId'] },
+    { required: ['operationPath'] },
+    { required: ['workflowId'] },
+  ],
+} as const;
+const steps = {
+  type: 'array',
+  items: step,
+} as const;
+const JSONSchema = {
+  type: 'object',
+  properties: {
+    type: {
+      type: 'string',
+      enum: ['object', 'array', 'string', 'number', 'integer', 'boolean', 'null'],
+    },
+    properties: {
+      type: 'object',
+      additionalProperties: true,
+    },
+    required: {
+      type: 'array',
+      items: { type: 'string' },
+    },
+    items: {
+      type: 'object',
+      additionalProperties: true,
+    },
+  },
+  required: ['type'],
+  additionalProperties: true,
+} as const;
+const workflow = {
+  type: 'object',
+  properties: {
+    workflowId: { type: 'string' },
+    summary: { type: 'string' },
+    description: { type: 'string' },
+    parameters: parameters,
+    dependsOn: { type: 'array', items: { type: 'string' } },
+    inputs: JSONSchema,
+    outputs: {
+      type: 'object',
+      additionalProperties: {
+        type: 'string',
+      },
+    },
+    steps: steps,
+    successActions: {
+      type: 'array',
+      items: onSuccessObject,
+    },
+    failureActions: {
+      type: 'array',
+      items: onFailureObject,
+    },
+  },
+  additionalProperties: false,
+  required: ['workflowId', 'steps'],
+} as const;
+const workflows = {
+  type: 'array',
+  items: workflow,
+} as const;
+export const workflowSchema = {
```

Comment:
Wouldn't it be better to rename it for consistency?

```suggestion
export const arazzoSchema = {
```